_posts/2020-05-14-beginners-config.md

```diff
@@ -0,0 +1,116 @@
+tagline: a Beginner’s Guide
+author: Laura Lorenz (Prefect), Julia Signell (Saturn Cloud)
+tags: [distributed, config]
+theme: twitter
+---
+{% include JB/setup %}
+
+
+_Configuring a Dask cluster can seem daunting at first, but the good news is that the Dask project has a lot of built in heuristics that try its best to anticipate and adapt to your workload based on the machine it is deployed on and the work it receives. Possibly for a long time you can get away with not configuring anything special at all. That being said, if you are looking for some tips to move on from using Dask locally, or have a Dask cluster that you are ready to optimize with some more in-depth configuration, these tips and tricks will help guide you and link you to the best Dask docs on the topic!_
+
+
+## How to host a distributed Dask cluster
+
+The biggest jump for me was from running a local version of Dask for just an hour or so at a time during development, to standing up a production-ready version of Dask. Broadly there are two styles:
+
+1. a static dask cluster -- one that is always on, always awake, always ready to accept work
+2. an ephemeral dask cluster -- one that is spun up or down easily with a Python API, and, when on, starts a minimal dask master node that itself only spins up dask workers when work is actually submitted
+
+Though those are the two broad main categories, there are tons of choices of how to actually achieve that. It depends on a number of factors including what cloud provider products you want to use and if those resources are pre-provisioned for you and whether you want to use a python API or a different deployment tool to actually start the Dask processes. A very exhaustive list of all the different ways you could provision a dask cluster is in the dask docs under [Setup](https://docs.dask.org/en/latest/setup.html). As just a taste of what is described in those docs, you could:
+
+*   Install and start up the dask processes [manually from the CLI](https://docs.dask.org/en/latest/setup/cli.html) on cloud instances you provision, such as AWS EC2 or GCP GCE
+*   Use popular deployment interfaces such as [helm for kubernetes](https://docs.dask.org/en/latest/setup/kubernetes-helm.html) to deploy dask in cloud container clusters you provision, such as AWS Fargate or GCP GKE
+*   Use ‘native’ deployment python APIs, provided by the dask developers, to create (and interactively configure) dask on deployment infrastructure they support, namely kubernetes with [dask-kubernetes](https://kubernetes.dask.org/en/latest/) and YARN with [dask-yarn](https://yarn.dask.org/en/latest/), as long as you’ve already provisioned the kubernetes cluster or hadoop cluster already
+*   Use a nearly full-service deployment python API called [Dask Cloud Provider](https://cloudprovider.dask.org/en/latest/), that will go one step farther and provision the cluster for you too, as long as you give it AWS credentials (and as of time of writing, it only supports AWS)
+
```

Comment:
Dask-Gateway should probably be in this list as well. https://github.com/dask/dask-sphinx-theme/issues/37